server/main.lua

```diff
@@ -262,38 +236,51 @@ exports('RegisterBossMenu', registerBossMenu)
 ---@param citizenid string
 ---@param job string
 local function doPlayerCheckIn(source, citizenid, job)
-	PlayersClockedIn[source] = { citizenid = citizenid, job = job }
-	onPlayerCheckIn(citizenid, job)
+	playersClockedIn[source] = { citizenid = citizenid, job = job }
+	OnPlayerCheckIn(citizenid, job)
 end
 
 ---@param source number
 local function onPlayerUnload(source)
-	if PlayersClockedIn[source] then
-        onPlayerCheckOut(PlayersClockedIn[source].citizenid, PlayersClockedIn[source].job)
-		PlayersClockedIn[source] = nil
+	if playersClockedIn[source] then
+        OnPlayerCheckOut(playersClockedIn[source].citizenid)
+		playersClockedIn[source] = nil
     end
 end
 
 ---@param source number
 RegisterNetEvent('QBCore:Server:OnPlayerLoaded', function()
-    local Player = exports.qbx_core:GetPlayer(source)
-	if Player ~= nil then
-		if Player.PlayerData.job.onduty then
-			doPlayerCheckIn(Player.PlayerData.source, Player.PlayerData.citizenid, Player.PlayerData.job.name)
-		end
+    local player = exports.qbx_core:GetPlayer(source)
+	if player == nil then return end
+	if player.PlayerData.job.onduty then
+		doPlayerCheckIn(player.PlayerData.source, player.PlayerData.citizenid, player.PlayerData.job.name)
+	end
+end)
+
+---@param source number
+---@param groupName string
+---@param groupGrade number
+AddEventHandler('qbx_core:server:onGroupUpdate', function(source, groupName, groupGrade)
```

Comment:
This would trigger when a group is changed even if it's not the primary one. I believe the event you'd want is https://docs.qbox.re/resources/qbx_core/events/client#qbcoreclientonjobupdate